nucleus/dataset.py

```diff
@@ -171,9 +171,9 @@ def annotate(
         if any((isinstance(ann, CuboidAnnotation) for ann in annotations)):
             raise NotImplementedError("Cuboid annotations not yet supported")
 
-        if asynchronous:
-            check_all_annotation_paths_remote(annotations)
+        check_all_annotation_paths_remote(annotations)
 
+        if asynchronous:
```

Comment:
Consider renaming `check_all_annotation_paths_remote` -> `check_all_annotation_paths_remote_mask` to make it clear this is only checking for `masks` rather than annotations in general.